# 인공지능수학: 통계학
# 숙제 13 (HW 13)


이름: 홍성규

학번: K2021035

# 아래 문제를 해결하는 코드를 작성하고, 그 결과를 쓰시오.

1. 어떤 농장에서 생산된 계란 35 개의 표본을
뽑아서 아래 변수 w에 저장하였다. 계란의 평균 무게에 대한 95% 신뢰 구간을
구하시오.

In [2]:
w=[22,
 24,
 23,
 22,
 21,
 27,
 26,
 27,
 25,
 29,
 21,
 25,
 28,
 26,
 20,
 21,
 25,
 20,
 28,
 23,
 26,
 20,
 21,
 27,
 24,
 28,
 25,
 26,
 20,
 24]

In [8]:
# 여기에 코드를 작성하시오.
import numpy as np
import scipy.stats as stats

xbar = np.mean(w)
s = np.std(w, ddof=1)
alpha = 0.05
zalpha = stats.norm.ppf(1-alpha/2)
v = zalpha * (s / np.sqrt(len(w)))
min = xbar - v
max = xbar + v
print('min : ', min, 'max : ', max)

min :  23.12675340021855 max :  25.139913266448115


(23.12675340021855, 25.139913266448115)

2. 어떤 정책에 대한 찬성율을 조사하기 위해 783명을
랜덤하게 선택하여 찬성 여부를 조사하였는데, 이 중 475 명이 찬성하였다. 찬성율 𝑝의 95% 신뢰구간 (confidence interval) 을 구하시오.

In [6]:
# 2.번 문제에 대한 코드
x = 475
n = 783
phat = x/n
alpha = 0.05
zalpha = stats.norm.ppf(1-alpha/2)
sd = np.sqrt((phat * (1-phat))/n)
ci = [phat - zalpha*sd, phat + zalpha*sd]
print(ci)

[0.5724252520070054, 0.640856995758001]


(0.5724252520070054, 0.640856995758001)

3. 위 1번의 문제에서 계란의 평균 무게는 23.5 그램으로 알려져 있다 . 새로운 사료를 도입한 후에 생산된 계란 35 개의 무게가 위 1번 문제의 w 변수에 저장된 것이라면, 새로운 사료가 평균적으로 더 무거운 계란을 생산한다고 할 수 있는가? 유의수준 5%로 검정하시오.

In [11]:
# 코드 작성

# H0:mu > 23.5 --> Z > zalpha

# 가설에 의한 모평균
mu = 23.5
# 표본평균
xbar = np.mean(w)
# 표본표준편차
sd = np.std(w, ddof=1)
z = (xbar-mu)/(sd/np.sqrt(len(w)))
print('검정통계량 : ', z)

# 유의수준
alpha = 0.05
# 임계값
cri = stats.norm.ppf(1-alpha)
print('임계값 : ', cri)

print(z > cri)

검정통계량 :  1.233196175191864
임계값 :  1.6448536269514722
False


(아니오)